src/client/debugger/jupyter/kernelDebugAdapter.ts

```diff
@@ -0,0 +1,331 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import * as vscode from 'vscode';
+import { DebugProtocol } from 'vscode-debugprotocol';
+import { randomBytes } from 'crypto';
+import * as path from 'path';
+import { IDebuggingCellMap, IJupyterSession } from '../../datascience/types';
+import { Kernel, KernelMessage } from '@jupyterlab/services';
+
+const debugRequest = (message: DebugProtocol.Request): KernelMessage.IDebugRequestMsg => {
+    return {
+        channel: 'control',
+        header: {
+            msg_id: randomBytes(8).toString('hex'),
+            date: new Date().toISOString(),
+            version: '5.2',
+            msg_type: 'debug_request',
+            username: 'vscode',
+            session: randomBytes(8).toString('hex')
```

Comment:
This might work for the spike, but for JMP messages session should be unique per session. It's not random per message. You can get the id of an existing session off the ISession on the IJupyterSession.